src/hooks/useDrafts/useDrafts.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react'
+import { addDraft, clearDrafts, getDraft, getDrafts, removeDraft, updateDraft } from './utils'
+
+export type CommonDraftProps = {
+  id: string
+  updatedAt?: string
+}
+
+export type DraftType = 'videos' | 'channels'
+
+export type VideoDraft = {
+  type: 'videos'
+  title?: string
+  description?: string
+  publicness?: 'public' | 'private'
+  publishedBefore?: string
+  marketing?: boolean
+  contentRating?: 'all' | 'mature'
+} & CommonDraftProps
+
+export type ChannelDraft = {
+  type: 'channels'
+  title?: string
+  description?: string
+  publicness?: 'public' | 'private'
+  language?: string
+} & CommonDraftProps
+
+export type Draft = VideoDraft | ChannelDraft
+
+type DraftState = {
+  videos: VideoDraft[]
+  channels: ChannelDraft[]
+}
+
+type ContextType = {
+  draftsState: DraftState
+  getInitialDrafts: () => Promise<void>
+}
+
+const DraftsContext = React.createContext<undefined | ContextType>(undefined)
+DraftsContext.displayName = 'DraftsContext'
+
+export const DraftsProvider: React.FC = ({ children }) => {
+  const [draftsState, setDraftsState] = useState<ContextType['draftsState']>({
+    channels: [],
+    videos: [],
+  })
+
+  const getInitialDrafts = useCallback(async () => {
```

Comment:
Not sure if `getInitialDrafts` is a good name. It makes sense within this context provider but we also use it in the hook to refetch the data, so at that point the drafts are no longer initial. Maybe just `fetchDrafts` or something along those lines?